src/features/user/Profile/components/RedeemModal.tsx

```diff
@@ -138,10 +136,19 @@ export default function RedeemModal({
     }
   }
 
+  useEffect(() => {
+    if (errorMessage || validCodeData) {
+      setAfterValidateCodeButton(true);
+    }
+  }, [errorMessage, validCodeData]);
+
```

Comment:
I think `afterValidateCode` isn't required to be set in the state. It is only being used to store the value of `errorMessage || validCodeData`. If you need a variable to store this, you can just use a regular variable (instead of storing it in state and triggering a rerender of the component).